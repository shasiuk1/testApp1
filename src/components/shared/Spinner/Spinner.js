/** @format */

import React from 'react';
import cssModules from 'react-css-modules';
import styles from './index.scss';

const Spinner = () => <div styleName="spinner" />;

Spinner.propTypes = {};

Spinner.defaultProps = {};

export default cssModules(Spinner, styles);
