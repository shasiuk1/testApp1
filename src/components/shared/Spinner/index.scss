@import '~SRC/styles/variables';
@import '~SRC/styles/mixins';

.spinner {
  display: inline-block;
  width: 1.5rem;
  height: 1.5rem;
  margin: 2px;
  vertical-align: text-bottom;
  border: .2em solid $text-color;
  border-right-color: transparent;
  border-radius: 50%;
  color: $text-color;
  animation: spinner-animate .75s linear infinite;
}


@keyframes spinner-animate {
  to { transform: rotate(360deg); }
}
