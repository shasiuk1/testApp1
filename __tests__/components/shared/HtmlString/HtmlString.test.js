/** @format */

import { shallow } from 'enzyme/build';
import React from 'react';
import Spinner from 'SRC/components/shared/Spinner';
import toJson from 'enzyme-to-json';

describe('test Spinner', () => {
  it('component is defined', () => {
    const wrapper = shallow(<Spinner />);
    expect(wrapper.find(Spinner)).toBeDefined();
  });

  it('component matches snapshot', () => {
    const wrapper = shallow(<Spinner />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
