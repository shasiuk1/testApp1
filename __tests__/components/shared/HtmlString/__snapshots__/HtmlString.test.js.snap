// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Spinner component matches snapshot 1`] = `
<div
  className="spinner"
/>
`;
